### ISSUE TITLE
Error message with modeltime::arima_reg and tidymodels rsample::fit_resample

### ISSUE THREAD

**Issue by wolff1234** [business-science/modeltime#48]
Hi Matt,

I am completely new to GitHub and I am trying to get an answer to the following problem I encountered. 

I am trying to use fit_resample() together with arima_reg() and I am getting an error message that I cannot understand. 

**I am using the following code:**

library(modeltime)
library(tidymodels)

df<-
data.frame(day=seq(as.Date("2020-1-1"), as.Date("2020-3-31"), "days"), 
value=sample(1:100, length(seq(as.Date("2020-1-1"), as.Date("2020-3-31"), "days"))))

splits<-
sliding_window(df, lookback = Inf, skip=6, assess_start = 7, assess_stop = 14, complete = T)

model_fit_arima <- 
arima_reg() %>% 
set_engine("auto_arima")%>% 
fit_resamples(value~day, data=df, resamples=splits)

**I am getting the following error message:**

x Slice01: model: Error: No date or date-time variable provided. Please supply a date or date-time variable as a...
x Slice02: model: Error: No date or date-time variable provided. Please supply a date or date-time variable as a...
x Slice03: model: Error: No date or date-time variable provided. Please supply a date or date-time variable as a...

**The same code works with other models, i.e. this works:**

 linear_reg() %>% set_engine("lm")%>% fit_resamples(value~day, data=df, resamples=splits)

**and the same data works with auto_arima without resampling, i.e. this works:** 

arima_reg() %>% set_engine("auto_arima")%>% fit(value~day,data=df)

**I am not sure, why _day_ is not accepted as date anymore, since:** 

class(analysis(splits$splits[[1]])$day)=="Date"
[1] TRUE

Many thanks in advances for your help and best wishes, Jan

------ 
**> sessionInfo()**
R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] yardstick_0.0.7    workflows_0.2.1    tune_0.1.1         rsample_0.0.8     
 [5] recipes_0.1.14     parsnip_0.1.4      modeldata_0.1.0    infer_0.5.3       
 [9] dials_0.0.9        scales_1.1.1       tidymodels_0.1.1   modeltime_0.3.0   
[13] RColorBrewer_1.1-2 skimr_2.1.2        forcats_0.5.0      purrr_0.3.4       
[17] tibble_3.0.3       tidyverse_1.3.0    fable_0.2.1        fabletools_0.2.0  
[21] tsibble_0.9.2      dummies_1.5.6      tseries_0.10-46    tscount_1.4.1     
[25] magick_2.0         gganimate_1.0.2    lubridate_1.7.4    fpp2_2.3          
[29] expsmooth_2.3      fma_2.3            forecast_8.5       readr_1.3.1       
[33] data.table_1.12.8  haven_2.3.1        DMwR_0.4.1         xgboost_1.2.0.1   
[37] circlize_0.4.5     plotROC_2.2.1      tableone_0.10.0    ggjoy_0.4.1       
[41] ggridges_0.5.1     stringr_1.4.0      cowplot_0.9.4      ranger_0.11.1     
[45] caret_6.0-81       pROC_1.16.2        rpart.plot_3.0.6   rpart_4.1-13      
[49] party_1.3-1        strucchange_1.5-1  sandwich_2.5-0     zoo_1.8-4         
[53] modeltools_0.2-22  mvtnorm_1.0-8      mctest_1.2         BSDA_1.2.0        
[57] lattice_0.20-38    tidyr_1.1.0        plotrix_3.7-4      XML_3.98-1.17     
[61] margins_0.3.23     broom_0.7.2        lme4_1.1-20        Matrix_1.2-15     
[65] foreign_0.8-72     ggpubr_0.2         magrittr_1.5       reshape2_1.4.3    
[69] ggplot2_3.3.2      readxl_1.3.1       dplyr_1.0.1        plyr_1.8.4        

loaded via a namespace (and not attached):
  [1] utf8_1.1.4           tidyselect_1.1.0     munsell_0.5.0        codetools_0.2-15    
  [5] future_1.11.1.1      withr_2.2.0          colorspace_1.4-1     knitr_1.29          
  [9] rstudioapi_0.11      ROCR_1.0-7           TTR_0.23-4           listenv_0.7.0       
 [13] repr_1.1.0           DiceDesign_1.8-1     farver_2.0.3         vctrs_0.3.2         
 [17] generics_0.0.2       TH.data_1.0-10       ipred_0.9-8          xfun_0.16           
 [21] R6_2.4.1             bitops_1.0-6         lhs_1.0.1            assertthat_0.2.1    
 [25] multcomp_1.4-8       nnet_7.3-12          gtable_0.3.0         globals_0.12.4      
 [29] timeDate_3043.102    rlang_0.4.7          GlobalOptions_0.1.0  splines_3.5.2       
 [33] ModelMetrics_1.2.2   yaml_2.2.1           prediction_0.3.6.2   abind_1.4-5         
 [37] modelr_0.1.6         backports_1.1.8      quantmod_0.4-13      tools_3.5.2         
 [41] lava_1.6.5           ltsa_1.4.6           ellipsis_0.3.1       gplots_3.0.1.1      
 [45] Rcpp_1.0.5           base64enc_0.1-3      progress_1.2.2       prettyunits_1.1.1   
 [49] slider_0.1.5         fracdiff_1.4-2       fs_1.5.0             survey_3.35-1       
 [53] furrr_0.1.0          warp_0.1.0           lmtest_0.9-36        reprex_0.3.0        
 [57] GPfit_1.0-8          hms_0.5.3            shape_1.4.4          compiler_3.5.2      
 [61] KernSmooth_2.23-15   crayon_1.3.4         minqa_1.2.4          StanHeaders_2.21.0-6
 [65] htmltools_0.5.0      RcppParallel_5.0.2   DBI_1.1.0            tweenr_1.0.1        
 [69] dbplyr_1.4.2         MASS_7.3-51.1        cli_2.0.2            quadprog_1.5-5      
 [73] gdata_2.18.0         parallel_3.5.2       gower_0.1.2          pkgconfig_2.0.3     
 [77] coin_1.2-2           xml2_1.3.2           foreach_1.4.4        prodlim_2018.04.18  
 [81] anytime_0.3.7        rvest_0.3.5          distributional_0.2.0 digest_0.6.25       
 [85] cellranger_1.1.0     curl_4.3             gtools_3.8.1         urca_1.3-0          
 [89] nloptr_1.2.1         lifecycle_0.2.0      nlme_3.1-137         jsonlite_1.7.0      
 [93] fansi_0.4.1          pillar_1.4.6         httr_1.4.1           survival_2.43-3     
 [97] glue_1.4.1           xts_0.11-2           iterators_1.0.10     class_7.3-14        
[101] stringi_1.4.6        caTools_1.17.1.1     e1071_1.7-0.1

**Comment by mdancho84** [comment]
## Problem

OK, I reproduced your error. The main issues is the __use of a formula as a preprocessor__. It's OK when you do this with `parsnip` model spec / fit, but with `workflows` formulas are treated inconsistently.  This is a known issue with `workflows` (see this tidymodels/workflows#70). 

I don't recommend using formulas as preprocessors with `workflow` objects because internally it does not respect date features. It converts them into numeric features, which ML algorithms can handle. Unfortunately, this is not good for time series modeling. 

## Solution 

Use a `recipe` as the preprocessor. 

## Working Example

A couple of points about this:

- __Sampling Procedure Not Acceptable for ARIMA:__ I noticed your use of `sliding_window()`. The parameters you selected had a large gap between the train/test. This is not good for an ARIMA model, which is a sequential model. Test set must immediately follow train set. I've updated the sampling procedure using `time_series_cv()`, which is what I teach in my [Time Series Course.](https://university.business-science.io/p/ds4b-203-r-high-performance-time-series-forecasting/) 

- __Use of Recipe as Preprocessor:__ Use a recipe instead of a formula preprocessor

- __Correlation Issue (Note/Warning)__: You will still get a warning because the ARIMA algorithm doesn't like the data. But it makes predictions. Solution - Use a different algorithm. This is easy to do with modeltime. Try `modeltime_fit_resamples()` from the `modeltime.resample` package. Tutorial Here: https://business-science.github.io/modeltime.resample/

- Learning recommendation: I think you'd benefit from taking my [High Performance Time Series Course](https://university.business-science.io/p/ds4b-203-r-high-performance-time-series-forecasting/). Honestly, it will teach you a ton. You'll go beyond ARIMA to Machine Learning & Deep Learning. Plus you'll learn how to tune models, ensemble, feature engineer, and much more.  

``` r
library(modeltime)
library(tidymodels)
library(tidyverse)
library(timetk)

set.seed(123)
df <- data.frame(
    day   = seq(as.Date("2020-1-1"), as.Date("2020-3-31"), "days"),
    value = sample(1:100, length(seq(as.Date("2020-1-1"), as.Date("2020-3-31"), "days")))
)

# splits <- sliding_window(df, lookback = Inf, skip=6, assess_start = 7, assess_stop = 14, complete = T)

splits <- df %>%
    time_series_cv(
        assess      = 14, 
        skip        = 7,
        cumulative  = TRUE,
        slice_limit = 6
    )
#> Using date_var: day

splits %>%
    tk_time_series_cv_plan() %>%
    plot_time_series_cv_plan(day, value, .facet_ncol = 2, .interactive = FALSE)
```

![](https://i.imgur.com/3EWwQcZ.png)

``` r

model_spec <- arima_reg() %>%
    set_engine("auto_arima")

resamples <- fit_resamples(
    object       = model_spec,
    # preprocessor = formula(value ~ day),
    preprocessor = recipe(value ~ day, splits$splits[[1]]),
    resamples    = splits,
    control      = control_resamples(save_pred = TRUE)
)
#> 
#> Attaching package: 'forecast'
#> The following object is masked from 'package:yardstick':
#> 
#>     accuracy
#> frequency = 7 observations per 1 week
#> ! Slice1: internal: A correlation computation is required, but `estimate` is const...
#> frequency = 7 observations per 1 week
#> ! Slice2: internal: A correlation computation is required, but `estimate` is const...
#> frequency = 7 observations per 1 week
#> ! Slice3: internal: A correlation computation is required, but `estimate` is const...
#> frequency = 7 observations per 1 week
#> ! Slice4: internal: A correlation computation is required, but `estimate` is const...
#> frequency = 7 observations per 1 week
#> ! Slice5: internal: A correlation computation is required, but `estimate` is const...
#> frequency = 7 observations per 1 week
#> ! Slice6: internal: A correlation computation is required, but `estimate` is const...

resamples %>%
    select(id, .predictions) %>%
    unnest(.predictions)
#> # A tibble: 84 x 4
#>    id     .pred  .row value
#>    <chr>  <dbl> <int> <int>
#>  1 Slice1  52.0    78     3
#>  2 Slice1  52.0    79    83
#>  3 Slice1  52.0    80    46
#>  4 Slice1  52.0    81    59
#>  5 Slice1  52.0    82    48
#>  6 Slice1  52.0    83    24
#>  7 Slice1  52.0    84    45
#>  8 Slice1  52.0    85    81
#>  9 Slice1  52.0    86    68
#> 10 Slice1  52.0    87    88
#> # … with 74 more rows
```

<sup>Created on 2020-11-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by wolff1234** [comment]
Hi Matt, okay, great, many thanks for your help. I've changed my approach and it works now! Many thanks, Jan￼
